src/xacro/__init__.py

```diff
@@ -431,11 +429,11 @@ def process_include(elt, macros, symbols, func):
             # restore filestack
             restore_filestack(oldstack)
```

Comment:
The `filestack` needs to be restored in any case, i.e. also in case of an exception.
Please use `finally` for this purpose.